src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/LinkedCasesMapping.java

```diff
@@ -20,17 +23,68 @@ private LinkedCasesMapping() {
 
     public static List<ServiceLinkedCases> getLinkedCases(SscsCaseData caseData) {
         return Optional.ofNullable(caseData.getLinkedCase())
-                .orElseGet(Collections::emptyList).stream()
-                .filter(Objects::nonNull)
-                .map(CaseLink::getValue)
-                .filter(Objects::nonNull)
+            .orElseGet(Collections::emptyList).stream()
+            .filter(Objects::nonNull)
+            .map(CaseLink::getValue)
+            .filter(Objects::nonNull)
+            .map(CaseLinkDetails::getCaseReference)
+            .filter(StringUtils::isNotBlank)
+            .map(caseReference -> ServiceLinkedCases.builder()
+                .caseReference(caseReference)
+                .caseName(HearingsCaseMapping.getPublicCaseName(caseData))
+                .reasonsForLink(HearingsCaseMapping.getReasonsForLink(caseData))
+                .build())
+            .collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public static List<ServiceLinkedCases> getLinkedCasesWithNameAndReasons(SscsCaseData caseData, CcdCaseService ccdCaseService) throws GetCaseException {
```

Comment:
This method feels like it should be split into two, one for getting linkedReferences, one for creating serviceLinkedCases